app/post.go

```diff
@@ -156,13 +156,14 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 
 	post.SanitizeProps()
 
-	pchan := make(store.StoreChannel)
+	var pchan store.StoreChannel
 	if len(post.RootId) > 0 {
-		go func(pchan store.StoreChannel) {
-			r, pErr := a.Srv.Store.Post().Get(post.RootId)
-			s := store.StoreResult{Data: r, Err: pErr}
-			pchan <- s
-		}(pchan)
+		postList, pErr := a.Srv.Store.Post().Get(post.RootId)
+		s := store.StoreResult{Data: postList, Err: pErr}
+		storeChan := make(store.StoreChannel, 1)
```

Comment:
Ok, you didn't get the point, my proposal would be more like:

```go
var pchan store.StoreChannel
if len(post.Rootid) > 0 {
	pchan = make(store.StoreChannel, 1)
	go func(pchan store.StoreChannel) {
		r, pErr := a.Srv.Store.Post().Get(post.RootId)
		pchan <- store.StoreResult{Data: r, Err: pErr}
		close(pchan)
	}(pchan)
}
```

Reply:
Actually, I understood your proposal. I saw your previous comment after i made my commit. I was waiting to check if build passes or not. But thanks for proposal.

Reply:
Oh sorry, I didn't realize that.

Reply:
no worries. @jespino tests are fixed.